contracts/EntryPoint.sol

```diff
@@ -281,9 +280,8 @@ contract EntryPoint is StakeManager {
      * revert with proper FailedOp in case paymaster reverts.
      * decrement paymaster's deposit
      */
-    function _validatePaymasterPrepayment(uint256 opIndex, UserOperation calldata op, bytes32 requestId, uint256 requiredPreFund, uint256 gasUsedByValidateWalletPrepayment) internal returns (bytes memory context) {
+    function _validatePaymasterPrepayment(uint256 opIndex, address paymaster, UserOperation calldata op, bytes32 requestId, uint256 requiredPreFund, uint256 gasUsedByValidateWalletPrepayment) internal returns (bytes memory context) {
```

Comment:
What is the point of doing it this way?

Reply:
avoiding reading the paymaster again from the calldata. I read it once (to know the payment mode) and then use it from memory. cheaper to pass it as parameter than reading it again from calldata. (solidity manages to cost ~200gas for each calldata member. it does a lot of useless boundary checks)